src/Storages/StorageMerge.cpp

```diff
@@ -965,8 +1132,14 @@ void ReadFromMerge::convertingSourceStream(
     const Aliases & aliases,
     ContextPtr local_context,
     QueryPipelineBuilder & builder,
-    const QueryProcessingStage::Enum & processed_stage)
+    QueryProcessingStage::Enum processed_stage,
+    RowPolicyData & row_policy_data)
 {
+    if (row_policy_data.needCare())
```

Comment:
`needCare()` is not a very clear name. `row_policy_data.hasRowPolicy()` seems better.

Reply:
Renamed, although it is usually considered bad practice if a method name contain class name (something like Vector::vector_size).
Nevertheless, let it be 'hasRowPolicy'.

Reply:
Better remove the method at all, see https://github.com/ClickHouse/ClickHouse/pull/50209#discussion_r1358108401